src/todos.rs

```diff
@@ -53,4 +54,17 @@ impl TodoStorage for Todos {
     fn get_collection(&self) -> &BTreeMap<u32, Todo> {
         &self.todo_collection
     }
+
+    fn resolve_one_todo(&mut self, key: u32) -> bool {
+        match self.get_one_todo(key) {
+            Some(todo) => self.update(
+                key,
+                Todo {
+                    message: todo.message.clone(),
+                    resolved: true,
+                },
+            ),
+            None => false,
+        }
```

Comment:
Não acho que é necessario tentar reusar os métodos já existentes aqui.

```suggestion
        let Some(todo) = self.todo_collection.get_mut(key) else {
            return false;
        };

        todo.resolved = true;

        true
```

Reply:
Ok! Usar o `get_mut` aqui ficou bem interessante por ele retornar um `&mut Todo`.